02a-dbc.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: page
+title: Testing
+subtitle: Design by Contract
+minutes: 15
+---
+> ## Learning Objectives {.objectives}
+>
+> *   Design by Contract is a way of using Assertions for interface specification.
+> *   Design by Contract defines pre-, post- and invariant conditions a function and a caller agree to obey.
+
+In [Design by Contract](https://en.wikipedia.org/wiki/Design_by_contract), the interaction between a caller
+and a library is treated as being governed by a *contract*. A contract involves three different types of requirements.
+
+* Pre-conditions: Things that must be true before calling the function.
+* Post-conditions: Things that are guaranteed to be true after the function returns.
+* Invariant-conditions: Things that are guaranteed not to change during the function's execution
+
+**Note**: In Python's contracts, only pre- and post-conditions are handled.
+
+In the examples here, we use [PyContracts](https://andreacensi.github.io/contracts/index.html#) which use
```

Comment:
Since "PyContracts" is just one library, I think this should be "which uses". The fact that PyContracts ends in an s doesn't make it a plural noun, really, in this case.